controllers/resources/templates.go

```diff
@@ -42,8 +42,21 @@ func (m *manager) GetTemplate(nhc *remediationv1alpha1.NodeHealthCheck) (*unstru
 
 // ValidateTemplates only returns an error when we don't know whether the template is valid or not, for triggering a requeue with backoff
 func (m *manager) ValidateTemplates(nhc *remediationv1alpha1.NodeHealthCheck) (valid bool, reason, message string, err error) {
-	template, templateError := m.GetTemplate(nhc)
-	valid, reason, message, err = m.validateTemplate(template, templateError)
+	if templateRef := nhc.Spec.RemediationTemplate; templateRef != nil {
+		template, templateError := m.GetTemplate(templateRef)
+		valid, reason, message, err = m.validateTemplate(template, templateError)
+		if !valid || err != nil {
+			return
+		}
```

Comment:
Nit: might be worth to refactor into a method in order to avoid code duplication

Reply:
outdated